File: src/services/NativeService/builtInTools.ts
import { invoke } from '@tauri-apps/api/core';

import type { BuiltInBashExecutionRequest, BuiltInBashExecutionResponse } from './types';

/**
 * 原生内置工具桥接层。
 */
export const builtInTools = {
    executeBash(request: BuiltInBashExecutionRequest): Promise<BuiltInBashExecutionResponse> {
        return invoke('built_in_tools_execute_bash', { request });
    },
} as const;

File: src-tauri/src/commands/built_in_tools.rs
// Copyright (c) 2026. 千诚. Licensed under GPL v3.

//! 内置工具原生命令。

use crate::core::built_in_tools::{
    self, BashExecutionRegistry, BuiltInBashExecutionRequest, BuiltInBashExecutionResponse,
};
use tauri::State;

/// 执行内置 Bash 工具请求。
///
/// 命令层保持薄封装，避免把参数校验、平台分支和进程生命周期管理
/// 散落到 Tauri 注册入口；真正的执行流程统一收口到核心层。
#[tauri::command]
pub async fn built_in_tools_execute_bash(
    request: BuiltInBashExecutionRequest,
    registry: State<'_, BashExecutionRegistry>,
) -> Result<BuiltInBashExecutionResponse, String> {
    built_in_tools::execute_bash(request, registry.inner()).await
}

/// 取消一条正在执行中的内置 Bash 请求。
#[tauri::command]
pub fn built_in_tools_cancel_bash(
    execution_id: String,
    registry: State<'_, BashExecutionRegistry>,
) -> Result<bool, String> {
    Ok(registry.cancel(&execution_id))
}
